synapse/lib/scope.py

```diff
@@ -179,5 +175,52 @@ def enter(vals=None):
     '''
     scope = _task_scope()
     scope.enter(vals)
-    yield
-    scope.leave()
+    try:
+        yield
+    finally:
+        scope.leave()
+
+def copy(task: asyncio.Task) -> None:
```

Comment:
Lets make this API named `clone` or something to differentiate it from the expectation of returning a copy